qiita_db/util.py

```diff
@@ -1219,3 +1219,159 @@ def supported_filepath_types(artifact_type):
+                    WHERE artifact_type='BIOM' AND
+                        study_id = qiita.study.study_id) AS artifact_biom_dts,
+                (SELECT array_agg(command_parameters ORDER BY artifact_id)
+                    FROM qiita.study_artifact
+                    LEFT JOIN qiita.artifact USING (artifact_id)
+                    LEFT JOIN qiita.artifact_type USING (artifact_type_id)
+                    WHERE artifact_type='BIOM' AND
+                        study_id = qiita.study.study_id)
+                    AS artifact_biom_params,
+                (SELECT array_agg(generated_timestamp ORDER BY artifact_id)
+                    FROM qiita.study_artifact
+                    LEFT JOIN qiita.artifact USING (artifact_id)
+                    LEFT JOIN qiita.artifact_type USING (artifact_type_id)
+                    WHERE artifact_type='BIOM' AND
+                        study_id = qiita.study.study_id) AS artifact_biom_ts,
+                (SELECT array_agg(DISTINCT visibility) FROM qiita.artifact
+                    LEFT JOIN qiita.visibility USING (visibility_id)
+                    WHERE study_id=qiita.study.study_id)
+                    AS artifacts_visibility,
+                (SELECT array_agg(publication_doi ORDER BY publication_doi)
+                    FROM qiita.study_publication
+                    WHERE study_id=qiita.study.study_id) AS publication_doi,
+                (SELECT array_agg(name ORDER BY email) FROM qiita.study_users
+                    LEFT JOIN qiita.qiita_user USING (email)
+                    WHERE study_id=qiita.study.study_id) AS shared_with_name,
+                (SELECT array_agg(email ORDER BY email) FROM qiita.study_users
+                    LEFT JOIN qiita.qiita_user USING (email)
+                    WHERE study_id=qiita.study.study_id) AS shared_with_email
+                FROM qiita.study
+                LEFT JOIN qiita.study_person ON (
+                    study_person_id=principal_investigator_id)
+                WHERE study_id IN %s"""
+        qdb.sql_connection.TRN.add(sql, [tuple(study_ids)])
+        infolist = []
+        refs = {}
+        for info in qdb.sql_connection.TRN.execute_fetchindex():
+            info = dict(info)
+
+            # publication info
+            if info['publication_doi'] is not None:
+                info['pmid'] = get_pubmed_ids_from_dois(
+                    info['publication_doi']).values()
+            else:
+                info['publication_doi'] = []
+                info['pmid'] = []
+
+            # visibility
+            info["status"] = infer_status([info['artifacts_visibility']])
+            del info['artifacts_visibility']
+
+            # pi info
+            info["pi"] = (info['pi_email'], info['pi_name'])
+            del info["pi_email"]
+            del info["pi_name"]
+
+            # shared with
+            info['shared'] = []
+            if info['shared_with_name']:
```

Comment:
Do you need the `if` statement? You should probably do the for loop directly, if it is an empty list it will not execute any iteration.

On the other hand, the `if` is over `shared_with_name` but from the for loop below it looks like the name may not exist. Should be better to do the if over `shared_with_email` which looks like is ensured to exist?

It is also possible that I'm missing something 😄